src/backend/joanie/tests/test_commands_generate_certificates.py

```diff
@@ -0,0 +1,273 @@
+        """
+        This command should accept three optional arguments:
+            - courses
+            - products
+            - orders
+        """
+        options = {
+            "courses": "00000",
+            "orders": uuid.uuid4(),
+            "products": uuid.uuid4(),
+        }
+
+        # TypeError: Unknown option(s) should not be raised
+        call_command("generate_certificates", **options)
+
+    def test_commands_generate_certificates(self):
+        """
+        If a certifying product contains graded courses with gradable course runs
+        and a user purchased this product and passed all gradable course runs,
+        a certificate should be generated
+        """
+
+        # Create a certifying product with one order eligible for certification
+        course_run = factories.CourseRunFactory(
+            enrollment_end=timezone.now() + timedelta(hours=1),
+            enrollment_start=timezone.now() - timedelta(hours=1),
+            is_gradable=True,
+            start=timezone.now() - timedelta(hours=1),
+        )
+        product = factories.ProductFactory(
+            price="0.00",
+            type=enums.PRODUCT_TYPE_CREDENTIAL,
+            target_courses=[course_run.course],
+        )
+        course = factories.CourseFactory(products=[product])
+        order = factories.OrderFactory(product=product, course=course)
+        certificate = models.Certificate.objects.filter(order=order)
+
+        self.assertEqual(certificate.count(), 0)
+
+        # DB queries should be minimized
+        call_command("generate_certificates")
+        self.assertEqual(certificate.count(), 1)
+
+        # But call it again, should not create a new certificate
+        call_command("generate_certificates")
+        self.assertEqual(certificate.count(), 1)
+
+    def test_commands_generate_certificates_can_be_restricted_to_order(self):
+        """
+        If `order` option is used, the review is restricted to it.
+        """
+        # Create a certifying product with two orders eligible for certification
+        course_run = factories.CourseRunFactory(
+            enrollment_end=timezone.now() + timedelta(hours=1),
+            enrollment_start=timezone.now() - timedelta(hours=1),
+            is_gradable=True,
+            start=timezone.now() - timedelta(hours=1),
+        )
+        product = factories.ProductFactory(
+            price="0.00",
+            type=enums.PRODUCT_TYPE_CREDENTIAL,
+            target_courses=[course_run.course],
+        )
+        course = factories.CourseFactory(products=[product])
+        orders = factories.OrderFactory.create_batch(2, product=product, course=course)
+        certificate = models.Certificate.objects.filter(order__in=orders)
+
+        self.assertEqual(certificate.count(), 0)
+
+        # A certificate should be generated for the 1st order
+        call_command("generate_certificates", order=orders[0].uid)
+        self.assertEqual(certificate.count(), 1)
```

Comment:
maybe add a filter clause'?
```suggestion
        self.assertEqual(certificate.filter(order=orders[0]).count(), 1)
```

If it works add it to all below tests.